java/arcs/core/analysis/PolicyVerifier.kt

```diff
@@ -70,49 +71,54 @@ class PolicyVerifier(val options: PolicyOptions) {
     }
 
     /**
-     * If there are any claims associated with this handle's stores, creates and returns a
-     * [InformationFlow.IngressInfo] after remapping the root of the claims from the store to this
-     * handle. Otherwise returns null.
+     * Generates a list of [InformationFlow.IngressInfo] for every ingress handle in the graph. The
+     * claims in the returned ingress info for a handle will be derived from the [PolicyConstraints]
+     * that apply to that handle. When there are no such constraints, an empty list of claims will
+     * be used. This will ensure that the data in the ingress handle cannot be egressed.
      */
-    private fun getIngressInfo(
-        handleNode: RecipeGraph.Node.Handle,
+    private fun getGraphIngresses(
+        graph: RecipeGraph,
         policyConstraints: PolicyConstraints
-    ): InformationFlow.IngressInfo? {
-        return policyConstraints.storeClaims[handleNode.handle.id]
-            ?.filterIsInstance<Claim.Assume>()
-            ?.map {
-                Claim.Assume(
-                    AccessPath(handleNode.handle, it.accessPath.selectors),
-                    it.predicate
-                )
+    ): List<InformationFlow.IngressInfo> {
+        return getIngressHandles(graph).map { handleNode ->
+            val partialClaims = handleNode.handle.type.toSchema().name?.name?.let { schemaName ->
+                policyConstraints.claims[schemaName]
             }
-            ?.let { claims -> InformationFlow.IngressInfo(handleNode, claims) }
-            // If there is information in `storeClaims`, use the safest information.
-            ?: getDefaultIngressInfo(handleNode)
+            val claims = partialClaims.orEmpty().map { it.inflate(handleNode.handle) }
```

Comment:
Should we only associate claims with handles that are written by particles explicitly marked as `@ingress`? 

The justification is that only stores that were created by an `@ingress` particle has policy associated with it and should be protected by policy. No operations would be allowed other "seemingly" ingress handles even if they match the `SchemaName` in the policy. 

I am not sure if that approach would be OK if we use a `map` handle as we won't see the corresponding ingress particle. 

WDYT?

Reply:
From our discussion offline: `map` handles will be treated as ingress points. "Seemingly" ingress handles will be an error.

Reply:
And from our follow up discussion: no error is needed for seemingly ingress handles. We're treating the `@isolated` annotation as sufficient indication from the developer that they know what they're doing, and so any output handles of such particles are not considered ingress points. 

Mapped handles are still considered ingress, and if their type matches the schema name mentioned in the policy, they will be treated as such.